src/v2/imports/carts/webjoint.ts

```diff
@@ -4,30 +4,32 @@ import { QueryStringContext } from "../../../shared/types";
 const webjointTracker = ({ appId, retailId }: Pick<QueryStringContext, "appId" | "retailId">) => {
   webjointDataSource({
     transactionEvent(transactionData) {
-      window.tracker(
-        "addTrans",
-        transactionData.id,
-        retailId ?? appId,
-        transactionData.total,
-        transactionData.tax,
-        transactionData.shipping,
-        transactionData.city,
-        transactionData.state,
-        transactionData.country
-      );
+      window.tracker("addTrans", {
+        orderId: transactionData.id,
+        affiliation: retailId ?? appId,
+        total: transactionData.total,
+        tax: transactionData.tax,
+        shippping: transactionData.shipping,
+        city: transactionData.city,
+        state: transactionData.state,
+        country: transactionData.country,
+        currency: transactionData.currency,
+      });
 
-      transactionData.items.forEach((item) => {
-        window.tracker(
-          "addItem",
-          item.orderId,
-          item.sku,
-          item.name,
-          item.category,
-          item.unitPrice,
-          item.quantity,
-          transactionData.currency
-        );
+      transactionData.items.forEach((items) => {
+        const { orderId, category, currency, name, quantity, sku, unitPrice } = items;
+        window.tracker("addItem", {
+          orderId,
+          sku,
+          name,
+          category,
+          unitPrice,
+          quantity,
+          currency,
+        });
```

Comment:
The correct schema for this is below:

```
snowplow('addItem', {
    orderId: '1234', // required
    sku: 'DD44',     // required
    name: 'T-Shirt',      
    category: 'Green Medium',
    price: 11.99,
    quantity: 1,
    currency: 'USD'
});
```

this will fail because `unitPrice` is not a valid field. Reference: https://docs.snowplow.io/docs/collecting-data/collecting-from-own-applications/javascript-trackers/javascript-tracker/javascript-tracker-v3/tracking-events/#ecommerce-tracking